test/FunctionalTests/Microsoft.OData.Client.Tests/Serialization/HttpClientRequestMessageTests.cs

```diff
@@ -18,5 +19,28 @@ public void UnwrapAggregateException()
             var exception = Assert.Throws<DataServiceTransportException>(() => msg.EndGetResponse(task));
             Assert.StartsWith("System.Exception: single exception\r\n", exception.Message);
         }
+
+#if NETCOREAPP2_0_OR_GREATER
+        [Fact]
+        public void SetCertificateValidationCallback()
+        {
+            var httpClientRequestMessage = new HttpClientRequestMessage(
+                new DataServiceClientRequestMessageArgs("GET", new Uri("http://localhost"), false, false, new Dictionary<string, string>()));
+
+            // set a predicate that is always true
+            httpClientRequestMessage.ServerCertificateCustomValidationCallback = (_, __, ___, ____) => true;
+            Assert.True(httpClientRequestMessage.ServerCertificateCustomValidationCallback(null, null, null, SslPolicyErrors.None));

```

Comment:
Is there a way to verify that these properties are actually set on the underlying HttpClientHandler?

Reply:
Alternatively, to verify that the callback is called when a request is made for which that callback is applicable?